client/js/lounge.js

```diff
@@ -111,6 +111,45 @@ $(function() {
 		index: 1
 	};
 
+	const foregroundColorStrategy = {
+		id: "foreground-colors",
+		match: /\x03(\d{0,2}|[A-Za-z ]{0,10})$/,
+		search(term, callback) {
+			term = term.toLowerCase();
+			const matchingColorCodes = constants.colorCodeMap
+				.filter(i => i[0].startsWith(term) || i[1].toLowerCase().startsWith(term));
+
+			callback(matchingColorCodes);
+		},
+		template(value) {
+			return `<span class="irc-fg${parseInt(value[0], 10)}">${value[1]}</span>`;
+		},
+		replace(value) {
+			return "\x03" + value[0];
+		},
+		index: 1
+	};
+
+	const backgroundColorStrategy = {
+		id: "background-colors",
+		match: /\x03(\d{2}),(\d{0,2}|[A-Za-z ]{0,10})$/,
+		search(term, callback, match) {
+			term = term.toLowerCase();
+			const matchingColorCodes = constants.colorCodeMap
+				.filter(i => i[0].startsWith(term) || i[1].toLowerCase().startsWith(term))
+				.map(pair => pair.concat(match[1])); // Needed to pass fg color to `template`...
+
+			callback(matchingColorCodes);
+		},
+		template(value) {
+			return `<span class="irc-fg${value[2]} irc-bg irc-bg${parseInt(value[0], 10)}">${value[1]}</span>`;
```

Comment:
Done. But don't you think we should be using `irc-fg04` and the likes instead of `irc-fg4`, to better reflect color code format in general?
That would remove need for `parseInt` here, but also I think it's nicer to match 1:1 with actual codes.
Not for this PR though, as they are currently used in a couple places I believe (which is why I wanted to use both until transition is over at least).